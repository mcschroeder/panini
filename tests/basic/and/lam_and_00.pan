and : (a:bool) -> (b:bool) -> {c:bool| c = true <=> (a = true /\ b = true)}

f00 = and

f01 = \x:bool. \y:bool. and x y


f10 : {x:bool|?} -> {y:bool|?} -> {z:bool|?}
f10 = and

f11 : {x:bool|?} -> {y:bool|?} -> {z:bool|?}
f11 = \x:bool. \y:bool. and x y


f20 : x:bool -> y:bool -> {z:bool|z = true <=> (x = true /\ y = true)}
f20 = and

f21 : x:bool -> y:bool -> {z:bool|z = true <=> (x = true /\ y = true)}
f21 = \x:bool. \y:bool. and x y
