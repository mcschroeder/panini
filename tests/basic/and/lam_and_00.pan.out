f00 : {a:𝔹 | true} → {b:𝔹 | true} → {c:𝔹 | c = true ⇔ a = true ∧ b = true}
f01 : {x:𝔹 | true} → {y:𝔹 | true} → {c:𝔹 | c = true ⇔ x = true ∧ y = true}
f10 : {x:𝔹 | true} → {y:𝔹 | true} → {z:𝔹 | z = true ⇔ x = true ∧ y = true}
f11 : {x:𝔹 | true} → {y:𝔹 | true} → {z:𝔹 | z = true ⇔ x = true ∧ y = true}
f20 : {x:𝔹 | true} → {y:𝔹 | true} → {z:𝔹 | z = true ⇔ x = true ∧ y = true}
f21 : {x:𝔹 | true} → {y:𝔹 | true} → {z:𝔹 | z = true ⇔ x = true ∧ y = true}